src/app/utils/field-validation/validators/dateValidator.ts

```diff
@@ -106,6 +110,77 @@ const makeDateValidator: DateValidatorConstructor = (dateField) => {
   }
 }
 
+/**
+ * Returns the number representation of a day of the week
+ */
+const convertInvalidDayToNumber = (invalidDay: InvalidDaysOptions): number => {
+  switch (invalidDay) {
+    case InvalidDaysOptions.Sunday:
+      return 0
+    case InvalidDaysOptions.Monday:
+      return 1
+    case InvalidDaysOptions.Tuesday:
+      return 2
+    case InvalidDaysOptions.Wednesday:
+      return 3
+    case InvalidDaysOptions.Thursday:
+      return 4
+    case InvalidDaysOptions.Friday:
+      return 5
+    case InvalidDaysOptions.Saturday:
+      return 6
+    default:
+      return -1
+  }
+}
+
+/**
+ * Convert the days of the week in the invalidDays array
+ * to a number array representing the number representation
+ * of the corresponding day of the week
+ */
+const convertInvalidDaysOfTheWeekToNumberSet = (
+  invalidDays: InvalidDaysOptions[],
+): Set<number> => {
+  if (invalidDays.length === 0) {
+    return new Set()
+  }
+
+  const invaliDaysNumberArray = invalidDays.map((invalidDay) =>
+    convertInvalidDayToNumber(invalidDay),
+  )
+  return new Set(invaliDaysNumberArray)
+}
+
+/**
+ * Returns a validator to check if date is an invalid day of the week.
+ */
+const makeInvalidDayOfTheWeekValidator: DateValidatorConstructor =
+  (dateField) => (response) => {
+    const { answer } = response
+
+    const invalidDaysOfTheWeekSet = convertInvalidDaysOfTheWeekToNumberSet(
+      dateField.invalidDays ?? [],
+    )
+
+    if (invalidDaysOfTheWeekSet.has(-1)) {
```

Comment:
nit: also suggest extracting `-1` into a constant instead of a magic number:

```ts
const INVALID_INVALID_DAY = -1 // The naming......
```

Reply:
Can I check if this fall through case is necessary, since we are guaranteed that the values in `invalidDay` array are of enum type `InvalidDaysOptions`?

Reply:
You're right. Remember to correctly type the constant then. 

Record<InvalidDayOption, number>